### PR TITLE
Patch/ranjanl/add sentencepiece requirement

### PR DESCRIPTION
**Pull request by snova-ranjanl**
Please refer to [CONTRIBUTING.md](CONTRIBUTING.md) for detailed instructions.

## Summary

- Adds sentencepiece
- Updates torch to compatible version
- Fixes test-paths to be more stable and library agnostic
- Udpates the way circleci-cache are generated
- Adds clarifying instructions for `pipenv` to docs

## PR Checklist
- [x] My PR is less than 500 lines of code
- [x] I have added sufficient comment as docstrings in my code
- [x] I have made corresponding changes to the documentation

- [ ] I have written unit-tests to test all of my code

### CODE REVIEW

**Review by snova-mattw — APPROVED** [review]
Approve for CI.

File: tests/test_multiprocessing.py [hunk 1]
```diff
@@ -14,19 +14,17 @@
...
 from generative_data_prep.data_prep import multiprocess_data_prep
-
-from .test_utils import EXAMPLE_PATH
+from tests.conftest import TESTS_EXAMPLES_PATH
 
 
 def get_split_dir(test_name: str) -> str:
     """Create a absolute path to example input."""
-    return os.path.join(Path.cwd(), EXAMPLE_PATH, test_name, f"pipelined_{test_name}", "splits")
+    return TESTS_EXAMPLES_PATH / test_name / f"pipelined_{test_name}" / "splits"
```
Comment (snova-zoltanc) [inline tests/test_multiprocessing.py:27]: Should we use / instead of os.path.join ? Would that break on some OS?

**Review by snova-zoltanc — APPROVED** [review]

File: tests/test_multiprocessing.py (hunk 1 above)
Comment (snova-ranjanl) [reply tests/test_multiprocessing.py:27]: `Path` is the most robust way to use define and access paths in Python; this should be compatible with all libraries

**Review by snova-zoltanc — APPROVED** [review]
LGTM, thanks again 😄

**Review by snova-mattw — APPROVED** [review]

**Pull request merged**